packages/web-components/fast-foundation/src/listbox/listbox.ts

```diff
@@ -177,16 +194,21 @@ export abstract class Listbox extends FoundationElement {
     }
 
     /**
-     * Focus the first selected option and scroll it into view.
+     * Ensures that the provided option is focused and scrolled into view.
      *
+     * @param optionToFocus - The option to focus
      * @internal
      */
-    protected focusAndScrollOptionIntoView(): void {
-        if (this.contains(document.activeElement) && this.firstSelectedOption) {
-            this.firstSelectedOption.focus();
-            requestAnimationFrame(() => {
-                this.firstSelectedOption.scrollIntoView({ block: "nearest" });
-            });
+    protected focusAndScrollOptionIntoView(
+        optionToFocus: ListboxOption | null = this.firstSelectedOption
+    ): void {
+        if (this.contains(document.activeElement)) {
+            if (optionToFocus) {
+                optionToFocus.focus();
+                requestAnimationFrame(() => {
```

Comment:
If you prefer, you could do `DOM.queueUpdate` here.

Reply:
I wasn't able to get it to work with `DOM.queueUpdate`. The timing needs to guarantee that `focus()` and `scrollIntoView()` happen on different frames, and since this function is called by `openChanged`, the `updateQueue` ends up grouping them into the same frame.

The scenario to test it is with the "Select with very long list" storybook example:

1. Press `Tab` to focus the select
2. Press `End` to select the last item in the list
3. Press `Space` to open the list

The listbox should be scrolled to the bottom and "Wyoming" should be focused.

Alternatively, waiting for `DOM.nextUpdate` works here:
```ts
optionToFocus.focus();
DOM.nextUpdate().then(() => {
    optionToFocus.scrollIntoView({ block: "nearest" });
});
```

Reply:
Good feedback here. Could we just add a comment for future proofing essentially saying this portion:

> The timing needs to guarantee that focus() and scrollIntoView() happen on different frames, and since this function is called by openChanged, the DOM.updateQueue ends up grouping them into the same frame, which is why we're using RAF